src/components/common/SearchBox.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import * as S from './SearchBox.styles'
+import ButtonReset from './ButtonReset'
+import PropTypes from 'prop-types'
+
+SearchBox.propTypes = {
+  register: PropTypes.func.isRequired,
+  onSubmit: PropTypes.func.isRequired,
+  reset: PropTypes.func.isRequired,
+  search: PropTypes.string.isRequired,
+}
+
+export default function SearchBox({ register, onSubmit, reset, search }) {
+  return (
+    <S.Form onChange={onSubmit} onSubmit={onSubmit} id="search">
```

Comment:
Looks strange that you forward to the same on function on change and submit

Reply:
without it my reset button didn't work because it's a button type='submit' to trigger the onSubmit function

Reply:
I see, maybe rename onSubmit to onSearch? As I understant it, your search is live and doesn't really care about the difference between onChange and onSubmit?